### PR TITLE
VDSL3: avoid calculating cache entries based on passthrough arguments.

### PR DESCRIPTION
**Pull request by DriesSchaumont**
## Issue ticket number and link
Closes #xxxx (Replace xxxx with the GitHub issue number)

## Checklist before requesting a review
- [x] I have performed a self-review of my code

- Does this PR contain:
  - [ ] Breaking changes
  - [ ] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [x] Bug fixes

- [x] Proposed changes are described in the CHANGELOG.md

- [ ] Relevant unit tests have been added

### CODE REVIEW

File: src/main/resources/io/viash/platforms/nextflow/VDSL3Helper.nf [hunk 1]
```diff
@@ -950,13 +958,10 @@ def workflowFactory(Map args) {
 
         def out = [ output[0], outputFiles ]
 
-        // passthrough additional items
-        if (output[1]) {
-          out.addAll(output[1])
-        }
-
         out
```
Comment (rcannood) [inline src/main/resources/io/viash/platforms/nextflow/VDSL3Helper.nf:961]: Might as well simplify to `[  output[0], outputFiles ]`

```suggestion
        [  output[0], outputFiles ]
```

File: src/main/resources/io/viash/platforms/nextflow/VDSL3Helper.nf [hunk 2]
```diff
@@ -825,7 +825,15 @@ def workflowFactory(Map args) {
...
       mid2_ = mid1_
     }
-    output_ = mid2_
+
+    output_passthrough = mid2_
+      | map { tuple ->
+        def id = tuple[0]
+        def passthrough = tuple.drop(2)
+        [id] + passthrough
+      }
```
Comment (rcannood) [inline src/main/resources/io/viash/platforms/nextflow/VDSL3Helper.nf:834]: Can simplify:

```suggestion
      | map { tuple ->
        // remove data from tuple
        [tuple[0]] + tuple.drop(2)
      }
```

**Review by rcannood — CHANGES_REQUESTED** [review]
Good work! 

Some suggestions:

File: src/main/resources/io/viash/platforms/nextflow/VDSL3Helper.nf [hunk 3]
```diff
@@ -950,13 +958,10 @@ def workflowFactory(Map args) {
...
 
-        // passthrough additional items
-        if (output[1]) {
-          out.addAll(output[1])
-        }
-
         out
       }
+
+    output_ = output_process.join(output_passthrough)
```
Comment (DriesSchaumont) [inline src/main/resources/io/viash/platforms/nextflow/VDSL3Helper.nf:964]: @rcannood should I add `failOnDuplicate` here?

Comment (rcannood) [reply src/main/resources/io/viash/platforms/nextflow/VDSL3Helper.nf:964]: Yes, good idea!

File: src/main/resources/io/viash/platforms/nextflow/VDSL3Helper.nf (hunk 1 above)
Comment (DriesSchaumont) [reply src/main/resources/io/viash/platforms/nextflow/VDSL3Helper.nf]: Indeed, adjusted!

File: src/main/resources/io/viash/platforms/nextflow/VDSL3Helper.nf (hunk 2 above)
Comment (DriesSchaumont) [reply src/main/resources/io/viash/platforms/nextflow/VDSL3Helper.nf]: Done!

File: src/main/resources/io/viash/platforms/nextflow/VDSL3Helper.nf (hunk 3 above)
Comment (DriesSchaumont) [reply src/main/resources/io/viash/platforms/nextflow/VDSL3Helper.nf]: Added!

**Review by rcannood — APPROVED** [review]
LGTM! (Assuming none of the tests fail, of course)

**Pull request merged**